SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -1428,6 +1430,27 @@ \section{Pipelines}
 
 \end{itemize}
 
+\section{The auto Keyword}
+\label{sec:TheAutoKeyword}
+
+Available since C++11, the \code{auto} keyword specifies that a variables type
+is automatically deduced from its initializer.
+
+The \code{auto} keyword should be used to specify a type in the following cases:
+
+\begin{itemize}
+\item The type is duplicated on the left side of an assigment when it is mandated on the
+      right side, i.e. when there is an explicit cast.
+\item When obtaining container elements, when the element type is obvious from the type
+      of the container.
+\item When declaring iterators in a \code{for} loop.
+\item When initializing with new, but not ITK's \code{::New()}.
```

Comment:
What's the issue with `auto filter = itkFilterType::New()`?
Isn't `filter` of type `itkFilterType::Pointer` ?

Reply:
@hjmjohnson suggested avoidance during `::New()` -- @hjmjohnson do you recall the logic for this?

Reply:
OK, I will remove it in that case.